File: apps/backend/app/schemas/verification_report.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class VerificationReportRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    workflow_id: UUID
    status: str
    passed: bool
    checks: list[dict[str, Any]] = Field(default_factory=list)
    warnings: list[dict[str, Any] | str] = Field(default_factory=list)
    errors: list[dict[str, Any] | str] = Field(default_factory=list)
    summary: str | None = None
    created_at: datetime

File: apps/frontend/src/types/verification-report.ts
export interface VerificationReport {
  id: string;
  workflow_id: string;
  status: string;
  passed: boolean;
  checks: Array<{ name: string; status: string; message: string; [key: string]: unknown }>;
  warnings: Array<Record<string, unknown> | string>;
  errors: Array<Record<string, unknown> | string>;
  summary?: string | null;
  created_at: string;
}
